executors/kafka/executor.go

```diff
@@ -0,0 +1,241 @@
+	MessagesFile string `json:"messages_file,omitempty" yaml:"messages_file,omitempty"`
+}
+
+// Result represents a step result.
+type Result struct {
+	Executor     Executor      `json:"executor,omitempty" yaml:"executor,omitempty"`
+	TimeSeconds  float64       `json:"timeSeconds,omitempty" yaml:"timeSeconds,omitempty"`
+	TimeHuman    string        `json:"timeHuman,omitempty" yaml:"timeHuman,omitempty"`
+	Messages     []Message     `json:"messages,omitempty" yaml:"messages,omitempty"`
+	MessagesJSON []interface{} `json:"messagesJSON,omitempty" yaml:"messagesJSON,omitempty"`
+	Err          string        `json:"error" yaml:"error"`
+}
+
+// ZeroValueResult return an empty implemtation of this executor result
+func (Executor) ZeroValueResult() venom.ExecutorResult {
+	r, _ := executors.Dump(Result{})
+	return r
+}
+
+// GetDefaultAssertions return default assertions for type exec
+func (Executor) GetDefaultAssertions() *venom.StepAssertions {
+	return &venom.StepAssertions{Assertions: []string{"result.err ShouldBeEmpty"}}
+}
+
+// Run execute TestStep of type exec
+func (Executor) Run(testCaseContext venom.TestCaseContext, l venom.Logger, step venom.TestStep, workdir string) (venom.ExecutorResult, error) {
+	var e Executor
+	if err := mapstructure.Decode(step, &e); err != nil {
+		return nil, err
+	}
+	start := time.Now()
+
+	result := Result{Executor: e}
+
+	if e.Timeout == 0 {
+		e.Timeout = 5000
+	}
+	if e.ClientType == "producer" {
+		err := e.produceMessages(workdir)
+		if err != nil {
+			result.Err = err.Error()
+		}
+	} else if e.ClientType == "consumer" {
+		var err error
+		result.Messages, result.MessagesJSON, err = e.consumeMessages(l)
+		if err != nil {
+			result.Err = err.Error()
+		}
+	} else {
+		return nil, fmt.Errorf("type must be a consumer or a producer")
+	}
+
+	elapsed := time.Since(start)
+	result.TimeSeconds = elapsed.Seconds()
+	result.TimeHuman = fmt.Sprintf("%s", elapsed)
+	result.Executor.Password = "****hidden****" // do not output password
+
+	return executors.Dump(result)
+}
+
+func (e Executor) produceMessages(workdir string) error {
+
+	if len(e.Messages) == 0 && e.MessagesFile == "" {
+		return fmt.Errorf("Missing messages or messagesFile configuration")
+	}
+
+	producerCfg := sarama.NewConfig()
+	producerCfg.Net.TLS.Enable = e.WithTLS // Enable TLS anyway
+	producerCfg.Net.SASL.Enable = e.WithSASL
+	producerCfg.Net.SASL.User = e.User
+	producerCfg.Net.SASL.Password = e.Password
+	producerCfg.Producer.RequiredAcks = sarama.WaitForLocal
+	producerCfg.Producer.Retry.Max = 10
+	producerCfg.Net.DialTimeout = 5 * time.Second
+	producerCfg.Producer.Return.Successes = true
+	producerCfg.Producer.Return.Errors = true
+	sp, err := sarama.NewSyncProducer(e.Addrs, producerCfg)
+	if err != nil {
+		return err
+	}
+	defer sp.Close()
+
+	messages := []*sarama.ProducerMessage{}
+
+	if e.MessagesFile != "" {
+		path := filepath.Join(workdir, string(e.MessagesFile))
+		if _, err := os.Stat(path); !os.IsNotExist(err) {
```

Comment:
Wrong condition, the *os.PathError returned by `Stat` can describes another type of error.